src/coreclr/vm/gc_unwind_x86.inl

```diff
@@ -0,0 +1,3890 @@
+// to specify some parameters for the above methods that are specific for a certain runtime
+// or configuration.
+#ifdef FEATURE_EH_FUNCLETS
+#define IN_EH_FUNCLETS(a) a
+#define IN_EH_FUNCLETS_COMMA(a) a,
+#else
+#define IN_EH_FUNCLETS(a)
+#define IN_EH_FUNCLETS_COMMA(a)
+#endif
+#ifdef FEATURE_NATIVEAOT
+#define NOT_IN_NATIVEAOT_PRE_COMMA(a)
+#else
+#define NOT_IN_NATIVEAOT_PRE_COMMA(a) ,a
+#endif
+
+#define RETURN_ADDR_OFFS        1       // in DWORDS
+
+#define X86_INSTR_TEST_ESP_SIB          0x24
+#define X86_INSTR_PUSH_0                0x6A    // push 00, entire instruction is 0x6A00
+#define X86_INSTR_PUSH_IMM              0x68    // push NNNN,
+#define X86_INSTR_W_PUSH_IND_IMM        0x35FF  // push [NNNN]
+#define X86_INSTR_CALL_REL32            0xE8    // call rel32
+#define X86_INSTR_W_CALL_IND_IMM        0x15FF  // call [addr32]
+#define X86_INSTR_NOP                   0x90    // nop
+#define X86_INSTR_NOP2                  0x9090  // 2-byte nop
+#define X86_INSTR_NOP3_1                0x9090  // 1st word of 3-byte nop
+#define X86_INSTR_NOP3_3                0x90    // 3rd byte of 3-byte nop
+#define X86_INSTR_NOP4                  0x90909090 // 4-byte nop
+#define X86_INSTR_NOP5_1                0x90909090 // 1st dword of 5-byte nop
+#define X86_INSTR_NOP5_5                0x90    // 5th byte of 5-byte nop
+#define X86_INSTR_INT3                  0xCC    // int3
+#define X86_INSTR_HLT                   0xF4    // hlt
+#define X86_INSTR_PUSH_EAX              0x50    // push eax
+#define X86_INSTR_PUSH_EBP              0x55    // push ebp
+#define X86_INSTR_W_MOV_EBP_ESP         0xEC8B  // mov ebp, esp
+#define X86_INSTR_POP_ECX               0x59    // pop ecx
+#define X86_INSTR_RET                   0xC2    // ret imm16
+#define X86_INSTR_RETN                  0xC3    // ret
+#define X86_INSTR_XOR                   0x33    // xor
+#define X86_INSTR_w_TEST_ESP_EAX        0x0485  // test [esp], eax
+#define X86_INSTR_w_TEST_ESP_DWORD_OFFSET_EAX   0x8485      // test [esp-dwOffset], eax
+#define X86_INSTR_w_LEA_ESP_EBP_BYTE_OFFSET     0x658d      // lea esp, [ebp-bOffset]
+#define X86_INSTR_w_LEA_ESP_EBP_DWORD_OFFSET    0xa58d      // lea esp, [ebp-dwOffset]
+#define X86_INSTR_w_LEA_EAX_ESP_BYTE_OFFSET     0x448d      // lea eax, [esp-bOffset]
+#define X86_INSTR_w_LEA_EAX_ESP_DWORD_OFFSET    0x848d      // lea eax, [esp-dwOffset]
+#define X86_INSTR_JMP_NEAR_REL32        0xE9    // near jmp rel32
+#define X86_INSTR_w_JMP_FAR_IND_IMM     0x25FF  // far jmp [addr32]
+
+#ifdef  _DEBUG
+// For dumping of verbose info.
+#ifndef DACCESS_COMPILE
+static  bool  trFixContext          = false;
+#endif
+static  bool  trEnumGCRefs          = false;
+static  bool  dspPtr                = false; // prints the live ptrs as reported
+#endif
+
+// NOTE: enabling compiler optimizations, even for debug builds.
+// Comment this out in order to be able to fully debug methods here.
+#if defined(_MSC_VER)
+#pragma optimize("tg", on)
+#endif
```

Comment:
TODO: This was moved unintentionally since it was in the middle of the x86 code. I'll move it back to eetwain.cpp.